src/network/networkd-wwan-bus.c

```diff
@@ -0,0 +1,1334 @@
+ *        to track modem plug/unplug
+ *      - /org/freedesktop/ModemManager1/Bearer org.freedesktop.DBus.Properties PropertiesChanged
+ *        to track bearers
+ * 2.3. Check if MM service is yet available: for that call /org/freedesktop/DBus, org.freedesktop.DBus
+ *      ListNames method and see if MM is available. If it is not, then wait for the NameOwnerChanged
+ *      signal and see when it is; finish initialization phase.
+ * 2.4. If MM is available - enumerate modems, see p.4.
+ * 2.5. Finish initialization phase.
+ *
+ * 3. Run-time
+ * 3.1. During the run-time we track MM service availability. When it is gone we remove all the modems
+ *      and bearers.
+ * 3.2. When MM is connected we do modem enumeration to get in sync with their current state.
+ * 3.3. If a modem was removed we also remove all its bearers.
+ * 3.4. If a modem was added we try to start a simple connect.
+ * 3.5. If connection was interrupted, e.g. modem changed its network connection from connected state
+ *      we start an automatic reconnect.
+ *
+ * 4. Modem enumeration
+ * 4.1. Modem enumeration is done by calling GetManagedObjects.
+ * 4.2. By receiving managed objects we try to instantiate all new modems found.
+ * 4.3. For that we inspect all bearers available for that modem and add all new bearers found.
+ * 4.4. We also read modem ports to detect WWAN interface name assigned to this modem, e.g. "wwan0" etc.
+ *      N.B. As we only get the interface name known that late and the corresponding .network file was
+ *      already used by the networkd to match interfaces etc. it is not possible
+ *      to do things like matching APN to .network and so on.
+ *
+ * 5. Simple (re)connect
+ * 5.1. Connection is done by calling org.freedesktop.ModemManager1.Modem.Simple Connect method for
+ *      the relevant modem.
+ * 5.2. It is possible that at the time of connect the operation may fail. For that reason and to ensure
+ *      we are always connected we employ a periodic timer which will re-try connection hoping it will
+ *      be successful this time or when modem has recovered after an error state and so on.
+ * 5.3. networkd will automatically start reconnection if any external entity disconnects modem from
+ *      the network.
+ *
+ * 6. Open questions
+ * 6.1. WWAN interfaces cannot be configured now with the existing .network file:
+ *      - ModemSimpleConnectProps - simple connect properties used to (re)connect: I've put
+ *        this under [Network] section for now. Is there a better name for that and location?
+ *      - RouteMetric - as we configure address/route dynamically (provided by the modem), we
+ *        do not use any section of .network to define/alter relevant parts. This patch uses
+ *        [DHCP]:RouteMetric for that for now.
+ *      - I have a use-case when I do not want the default gateway to be set for WWAN inteface.
+ *        I use [DHCPv4]:UseGateway now which probably needs a better way of defining.
```

Comment:
This is outdated now? Please update.
Also, this is a kind of internal document. So, `I have ...` or `I do not want ...` sounds spurious.